bundles/org.openhab.binding.bluetooth.generic/src/main/java/org/openhab/binding/bluetooth/generic/internal/GenericBluetoothHandler.java

```diff
@@ -84,20 +85,51 @@ public void initialize() {
         readCharacteristicJob = scheduler.scheduleWithFixedDelay(() -> {
             if (device.getConnectionState() == ConnectionState.CONNECTED) {
                 if (resolved) {
-                    for (CharacteristicHandler charHandler : charHandlers.values()) {
```

Comment:
I'd prefer if you kept this old iteration method and used `charHandler.characteristic` to get the characteristic instead. Creation of `CharacteristicHandler` shouldn't be done in this part of the code.

Reply:
Is there a better place to go through all characteristics so that we can check if they are able to notify, check if the channel is linked and then subscribe to it if applicable?

Reply:
In my own branch I had to create a new map in order to lookup the channels for a given characteristic.

I haven't done much work on my branch in a while to feel free to to cherry pick my changes in GenericBluetoothHandler:
https://github.com/cpmeister/openhab-addons/blob/7855a238e02739a91d2583b0cf56e0f3f3d1560f/bundles/org.openhab.binding.bluetooth.generic/src/main/java/org/openhab/binding/bluetooth/generic/internal/GenericBluetoothHandler.java#L69

Reply:
Great thanks, I look into that 👍